File: article_35/04_immutable_strings.c
/*
 Seriál "Programovací jazyk Go"

 Třicátá pátá část
    Programovací jazyk Go pro skalní céčkaře (2.část)
    https://www.root.cz/clanky/programovaci-jazyk-go-pro-skalni-ceckare-2-cast/
*/

#include <stdio.h>

int main(void) {
    const char *s = "www.root.cz";

    puts(s);
    s[3] = '*';
    s[8] = '*';
    puts(s);

    return 0;
}

File: article_35/04_immutable_strings.go
// Seriál "Programovací jazyk Go"
//    https://www.root.cz/serialy/programovaci-jazyk-go/
//
// Třicátá pátá část
//    Programovací jazyk Go pro skalní céčkaře (2.část)
//    https://www.root.cz/clanky/programovaci-jazyk-go-pro-skalni-ceckare-2-cast/
//
// Repositář:
//    https://github.com/tisnik/go-root/
//
// Seznam demonstračních příkladů ze třicáté páté části:
//    https://github.com/tisnik/go-root/blob/master/article_35/README.md
//
// Demonstrační příklad číslo 4:
//     Řetězce jsou v Go neměnitelné.

package main

import "fmt"

func main() {
	var s string = "www.root.cz"

	fmt.Println(s)
	s[3] = '*'
	s[8] = '*'
	fmt.Println(s)
}
